src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageCommon.cpp

```diff
@@ -108,7 +108,7 @@ ListBlobsPagedResponse ContainerClientWrapper::ListBlobs(const ListBlobsOptions
     new_options.Prefix = blob_prefix / options.Prefix.ValueOr("");
 
     auto response = client.ListBlobs(new_options);
-    auto blob_prefix_str = blob_prefix.empty() ? "" : blob_prefix.string() + "/";
+    auto blob_prefix_str = blob_prefix.empty() ? "" : (blob_prefix.string() + (blob_prefix.string().back() == '/' ? "" : "/"));
```

Comment:
I think we can use `fs::path(blob_prefix) / ""` here. It handles all three cases: https://godbolt.org/z/hT8sMMPEd.